cg/apps/downsample/downsample.py

```diff
@@ -0,0 +1,169 @@
+            family=case,
+            sample=sample,
+            status=self.downsample_data.sample_status(sample=sample),
+        )
+        self.status_db.session.add(sample_case_link)
+        self.status_db.session.commit()
+        LOG.info(f"Related sample {sample.internal_id} to {case.internal_id}")
+
+    def add_downsampled_sample_case_to_statusdb(self) -> None:
+        """
+        Add the downsampled sample and case to statusDB and generate the sample case link.
+        """
+        self.add_downsampled_sample_entry_to_statusdb()
+        self.add_downsampled_case_to_statusdb()
+        self._link_downsampled_sample_to_case(
+            sample=self.downsample_data.downsampled_sample,
+            case=self.downsample_data.downsampled_case,
+        )
+
+    def is_decompression_needed(self) -> bool:
+        """Check if decompression is needed for the specified case.
+        Decompression is needed if there are no files with fastq tag found for the sample in housekeeper.
+        """
+        LOG.debug("Checking if decompression is needed.")
+        is_decompression_needed: bool = False
+        if not self.housekeeper_api.get_files(
+            bundle=self.downsample_data.original_sample.internal_id,
+            tags=[SequencingFileTag.FASTQ],
+        ):
+            is_decompression_needed = True
+        return is_decompression_needed
+
+    def start_decompression(self, sample: Sample) -> None:
+        """Start decompression of spring compressed fastq files for the given sample."""
+        LOG.info(f"Decompression for {sample.internal_id} is needed.")
+        self.prepare_fastq_api.compress_api.decompress_spring(sample.internal_id)
+        LOG.info("Re-run down sample command when decompression is done.")
+
+    def start_downsample_job(self, original_sample: Sample, sample_to_downsample: Sample) -> int:
+        """
+        Start a down sample job for a sample.
+        -Retrieves input directory from the latest version of a sample bundle in housekeeper
+        -Starts a down sample job
+        """
+        downsample_work_flow = DownsampleWorkflow(
+            number_of_reads=multiply_by_million(self.downsample_data.number_of_reads),
+            config=self.config,
+            output_fastq_dir=str(self.downsample_data.fastq_file_output_directory),
+            input_fastq_dir=str(self.downsample_data.fastq_file_input_directory),
+            original_sample=original_sample,
+            downsampled_sample=sample_to_downsample,
+            dry_run=self.dry_run,
+        )
+        return downsample_work_flow.write_and_submit_sbatch_script()
+
+    def downsample_sample(self) -> int | None:
+        """Down sample a sample."""
+        if self.is_decompression_needed():
+            self.start_decompression(self.downsample_data.original_sample)
+            return
+        LOG.debug("No Decompression needed.")
+        self.add_downsampled_sample_case_to_statusdb()
```

Comment:
I'm not a big fan of having database names in methods, but does not really matter (which database it writes to could be considered an implementation detail that is irrelevant to the caller). Feel free to keep it though, we do it everywhere